providers/gupshup/src/lib/gupshup.provider.ts

```diff
@@ -43,16 +46,14 @@ export class GupshupSmsProvider implements ISmsProvider {
       }),
     };
 
-    const opts = {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(params),
-    };
+    const response = await axios.post(GupshupSmsProvider.BASE_URL, params, {
+      httpsAgent: new Agent({
+        rejectUnauthorized: false,
+        secureOptions: crypto.constants.SSL_OP_LEGACY_SERVER_CONNECT,
```

Comment:
@RahulDutta007,
Why is this change required?

cc: @LetItRock

Reply:
I was getting the fetch error "ERR_SSL_UNSAFE_LEGACY_RENEGOTIATION_DISABLED". Possibly due to running it as a localhost as from Node v18, it does not allow legacy TLS renegotiation and hence, unable to test.

Reply:
@RahulDutta007, can you try using`127.0.0.1` instead of `localhost`?
I think we wont need this change.